contracts/core/src/xchain/OmniPortal.sol

```diff
@@ -173,6 +173,7 @@ contract OmniPortal is
         XTypes.BlockHeader calldata xheader = xsub.blockHeader;
         uint64 valSetId = xsub.validatorSetId;
 
+        require(xheader.consensusChainId == omniCChainId, "OmniPortal: wrong consensus chain");
```

Comment:
Can we make this reason string one character smaller? It's at 33 we want 32 so it fits into a single 32 byte slot.

Reply:
And let's add a test case

Reply:
Done, and done! Thank you!